parsers/src/main/scala/uk/gov/ons/addressIndex/crfscala/CrfParser.scala

```diff
@@ -0,0 +1,120 @@
+package uk.gov.ons.addressIndex.crfscala
+
+import uk.gov.ons.addressIndex.crfscala.CrfScala._
+
+//TODO scaladoc
+trait CrfParser {
+
+  //TODO scaladoc
+  val crfScala: CrfScalaJniImpl = new CrfScalaJniImpl()
+
+  //TODO scaladoc
+  def tag(i: Input, fas: CrfFeatures, tokenable: CrfTokenable): String = {
+    val currentDirectory = new java.io.File(".").getCanonicalPath
+    val modelPath = s"$currentDirectory/parsers/src/main/resources/addressCRFA.crfsuite"
+    val actual = parse(i, fas, tokenable)
+    val augmentedActual = actual
+      .split(CrfScalaJni.lineEnd)
+      .map(
+        _.split(CrfScalaJni.delimiter)
+         .sorted
+         .mkString(CrfScalaJni.delimiter)
+      )
+      .mkString(CrfScalaJni.lineEnd) + CrfScalaJni.lineEnd
+    crfScala.tag(modelPath, augmentedActual)
+
+  }
+
+  //TODO scaladoc
+  def parse(i: Input, fas: CrfFeatures, tokenable: CrfTokenable): CrfJniInput = {
+    val tokens = tokenable(i)
+    val preprocessedTokens = tokenable normalise tokens
+    val onlyOneToken = preprocessedTokens.length == 1
+    val onlyTwoTokens = preprocessedTokens.length == 2
+    val multipleTokens = preprocessedTokens.length > 2
+    val sb = StringBuilder.newBuilder
+
+    if(onlyOneToken) {
```

Comment:
This is a bit of a hack due to time. I have a Aggregate Feature Analyser branch which solves all this